components/homepage/HomeScreen.js

```diff
@@ -0,0 +1,129 @@
+import * as React from 'react';
+import { List, Checkbox } from 'react-native-paper';
+import { SafeAreaView } from 'react-native-safe-area-context';
+import { useSelector, useDispatch } from 'react-redux';
+import { View, Text, Button, TouchableOpacity, ScrollView } from 'react-native';
+import { fetchAllTasks, updateTasks } from '../../redux/TasksSlice';
+import { fetchAllTmrTasks, updateTmrTasks } from '../../redux/TomorrowTasksSlice';
+import { Bars3CenterLeftIcon, PlusCircleIcon } from "react-native-heroicons/solid";
+import { colors } from '../../theme/colors';
+import { formatTime } from '../../helper/TimeFormat';
+
+const HomeScreen = ({ navigation }) => {
+
+    const dispatch = useDispatch();
+
+    const tasks = useSelector(state => state.tasks.listTasks);
+    const tmrTasks = useSelector(state => state.tmrTasks.listTasks);
+
+    const _handleCheckbox = (taskId) => {
+        // dispatch(markDone({ id: taskId }))
+        // console.log('tick')
+        const newListTasks = tasks.concat()
+        const selectedTask = tasks.find(task => task.id === taskId);
+        const index = newListTasks.findIndex(task => task.id === taskId)
+        newListTasks[index] = {
+            id: selectedTask.id, name: selectedTask.name,
+            time: selectedTask.time, isDone: !selectedTask.isDone
+        }
```

Comment:
These logic should be done in the reducer, we just need to dispatch selected task id then the actual logic will stay in the reducer